packages/util/test/Percent.test.ts

```diff
@@ -0,0 +1,56 @@
+import { calcPercentages } from '../src/Percent';
+
+describe('Percent', () => {
+  it('single value is always 100%', () => {
+    expect(calcPercentages([50])).toEqual([1]);
+  });
+
+  it('whole percentages add up to 100%', () => {
+    expect(calcPercentages([50, 50])).toEqual([0.5, 0.5]);
+  });
+
+  it('floating point percentages', () => {
+    expect(calcPercentages([403, 597])).toEqual([0.403, 0.597]);
+    expect(calcPercentages([249, 249, 502])).toEqual([0.249, 0.249, 0.502]);
+    expect(calcPercentages([255, 245, 265, 235])).toEqual([0.255, 0.245, 0.265, 0.235]);
+  });
+
+  it('percentages smaller than 1%', () => {
+    expect(calcPercentages([5, 6, 1000 - 5 - 6])).toEqual([0.005, 0.006, 0.989]);
+    expect(calcPercentages([0, 6, 1000 - 6])).toEqual([0, 0.006, 0.994]);
+  });
+
+  it('one part is zero, total is implicitly zero, so it takes 100% of the total', () => {
+    expect(calcPercentages([0])).toEqual([1]);
+  });
+
+  it('multiple parts are zero, total is implicitly zero, percent is distributed evenly', () => {
+    expect(calcPercentages([0, 0, 0, 0])).toEqual([0.25, 0.25, 0.25, 0.25]);
+  });
+
+  it('total is adjusted to equal at least as much as the sum of the parts', () => {
```

Comment:
This seems a little odd to me. Is there a use case? If not, perhaps throwing would be better?

Reply:
No use case comes to mind. I added it to guard against abnormal inputs.
It could happen if the parts balance calculator emits before the total calculator, depending on how the observables are combined, that there is an invocation to calculate percentages where the total hasn't caught up with the balances.
If that's alright, I would be a bit defensive in this case.